slang-gfx.h

```diff
@@ -1282,6 +1362,11 @@ class IRenderCommandEncoder : public ICommandEncoder
         return rootObject;
     }
 
+    // Sets the current pipeline state and bind an existing root shader object.
+    virtual SLANG_NO_THROW Result SLANG_MCALL bindPipelineAndRootObject(
+        IPipelineState* state,
+        IShaderObject* rootObject) = 0;
```

Comment:
The biggest question(s) I have are all around this operation and what its semantics are.

I assume that `rootObject` needs to have been allocated using `createMutableRootShaderObject`, and this operation should be an error if it wasn't.

Am I correct to assume that while a pre-existing root shader object like this is bound any changes made to the object will automatically be mirrored to the active pipeline state (used by the next draw/dispatch/whatever)? That would make sense given how the existing root shader object case works, but I'd want to confirm.

Reply:
Yes, that is the intention here.